server/src/main/java/org/apache/iotdb/db/qp/strategy/LogicalGenerator.java

```diff
@@ -1167,31 +1176,52 @@ private FilterOperator parsePredicate(PredicateContext ctx) {
       return parseOrExpression(ctx.orExpression());
     } else {
       Path path = null;
```

Comment:
If path = null, can we throw a self-defined exception instead of NullPointer exception?

Reply:
according to the anltr4 gramma `(TIME | TIMESTAMP | suffixPath | prefixPath)`.  `path == null` is impossible.